lib/liquid/drop.rb

```diff
@@ -24,8 +24,9 @@ class Drop
     attr_writer :context
 
     # Catch all for the method
-    def liquid_method_missing(_method)
-      nil
+    def liquid_method_missing(method)
+      return nil unless @context.strict_variables
```

Comment:
@gauravmc (regarding https://github.com/Shopify/liquid/pull/691#issuecomment-188461364) more specifically a drop doesn't always have the `@context` set.  This is the line that was causing the NoMethodError exceptions.